handlers/update_resource.go

```diff
@@ -39,12 +43,25 @@ func (d *updateResourceEntry) Handle(params operations.UpdateResourceParams) mid
 		panic(err)
 	}
 
-	merged := d.mergeJSON(&existingResource.JSON, &newResource.JSON)
-	newResource = datautils.NewResource(merged).
-		WithExternalIdentifier(id).   // Don't allow changing druids
-		WithID(existingResource.ID()) // Ignore any passed in tacoIdentifier
+	err = d.verifyPayload(newResource, existingResource)
+	if err != nil {
+		return operations.NewUpdateResourceUnprocessableEntity()
+	}
+
+	v, _ := newResource.JSON["version"].(json.Number).Float64()
+	if int(v) > existingResource.Version() {
+		d.buildNewResourceVersion(newResource, existingResource)
```

Comment:
can you pass `v` into this func, so that we don't duplicate line 51 on line 104?